ui/static/ui/js/manage_taxonomies.jsx

```diff
@@ -314,9 +323,51 @@ define('manage_taxonomies', ['react', 'lodash', 'jquery', 'utils'],
     getInitialState: function() {
       return {
         vocabularies: [],
-        learningResourceTypes: []
+        learningResourceTypes: [],
+        message: ''
       };
     },
+    deleteVocabulary: function(vocab, dialogId) {
+      if (this.isMounted() && vocab && dialogId) {
+        var options = {
+          tag: vocab,
+          confirmationDialogId: dialogId,
+          confirmationDialogActionButtonName: "Delete",
+          actionButtonClass: "btn btn-danger btn-ok",
+          confirmationDialogTitle: "Confirm Delete",
+          confirmationDialogMessage:
+          "Are you sure you want to delete vocabulary" +
+          " '" +  vocab.name + "'.",
+          confirmationSuccess: this.confirmationDeleteResponse
+        };
+        this.props.renderConfirmationDialog(options);
+      }
+    },
+    confirmationDeleteResponse: function(status, vocab) {
+      var method = "DELETE";
+      var thiz = this;
+      if (status && vocab) {
+        var API_ROOT_VOCAB_URL = '/api/v1/repositories/' + this.props.repoSlug +
+        '/vocabularies/' + vocab.slug;
+        $.ajax({
+          type: method,
+          url: API_ROOT_VOCAB_URL
+        }).fail(function() {
+          var message = "Unable to delete vocabulary '" + vocab.name + "'";
+          thiz.setState({message: {error: message}});
+        }).done(function() {
+          var editIndex = -1;
+          var vocabularies = thiz.state.vocabularies;
+          editIndex = _.findIndex(vocabularies, function(vocabularyObj) {
+            return vocabularyObj.vocabulary.id === vocab.id;
+          });
+          if (editIndex > -1) {
+            vocabularies.splice(editIndex, 1);
```

Comment:
`vocabularies` comes from `this.state` so it cannot be altered, you must make a copy of it first. Instead of using `findIndex` to find the position and using `splice` to remove the item, you can make a copy using `_.filter` instead.